addons/website_sale/models/website.py

```diff
@@ -633,3 +638,62 @@ def _get_checkout_steps(self, current_step=None):
     def has_ecommerce_access(self):
         """ Return whether the current user is allowed to access eCommerce-related content. """
         return not (self.env.user._is_public() and self.ecommerce_access == 'logged_in')
+
+    def configurator_addons_apply(self, **kwargs):
+        def _generate_ai_ecommerce_categories(industry_name):
+            """
+            Generate e-commerce categories for a given industry using ChatGPT API.
+            """
+
+            prompt = f"""
+            My business is {industry_name} and I am building an eCommerce website.
+            Create 8 categories with a name and a description. It has to be persuasive and attract the attention of my visitors but keep it very short and to the point.
+            Provide the answer as a formatted json.
+            """
+            try:
+                IrConfigParameter = self.env['ir.config_parameter'].sudo()
+                olg_api_endpoint = IrConfigParameter.get_param(
+                    'web_editor.olg_api_endpoint', DEFAULT_OLG_ENDPOINT
+                )
+                database_id = IrConfigParameter.get_param('database.uuid')
+                response = iap_tools.iap_jsonrpc(olg_api_endpoint + "/api/olg/1/chat", params={
+                    'prompt': prompt,
+                    'conversation_history': [],
+                    'database_id': database_id,
+                }, timeout=30)
+                if response['status'] == 'success':
+                    try:
+                        content = response['content'].replace('```json\n', '').replace('\n```', '')
+                        return json.loads(content)
+                    except json.JSONDecodeError:
+                        logger.warning("API response is not a valid JSON for the category generation")
+                        return False
+                elif response['status'] == 'error_prompt_too_long':
+                    logger.warning("Prompt is too long for the category generation")
+                elif response['status'] == 'limit_call_reached':
+                    logger.warning("Limit call reached for the category generation")
+                else:
+                    logger.warning("Response could not be generated for the category generation")
+            except AccessError:
+                logger.warning("API is unreachable for the category generation")
+            return False
+
+        industry_name = kwargs.get('industry_name')
+        if industry_name and 8 in kwargs.get('selected_features'):  # If the shop feature is selected
```

Comment:
What about defining a constant `SHOP_FEATURE_ID = 8` ?
Especially given that there are actual numeric 8 used in the same method :grimacing: